File: src/main/java/ar/edu/unq/unidad3/modelo/Mago.java
package ar.edu.unq.unidad3.modelo;

import lombok.*;

import jakarta.persistence.*;

@Getter @Setter @NoArgsConstructor @ToString

@Entity
public final class Mago extends Personaje {
    private Integer magia;

    public Mago(@NonNull String nombre, @NonNull Integer vida, @NonNull Integer pesoMaximo, @NonNull Integer magia) {
        this.setNombre(nombre);
        this.setVida(vida);
        this.setPesoMaximo(pesoMaximo);
        this.setMagia(magia);
    }
}

File: src/main/kotlin/ar/edu/unq/unidad3/modelo/Mago.kt
package ar.edu.unq.unidad3.modelo

import ar.edu.unq.unidad3.modelo.exception.MuchoPesoException
import javax.persistence.*
import kotlin.collections.HashSet

@Entity
class Mago() : Personaje() {
    var magia: Int = 0

    constructor(nombre: String) : this() {
        this.nombre = nombre
    }
}
